src/main/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -354,7 +357,7 @@ protected int doWork() {
                     rec.getReadName().equals(opticalDuplicateQueryName) ||
                     recordInFileIndex == nextOpticalDuplicateIndex;
 
-            rec.setAttribute(DUPLICATE_TYPE_TAG, null);
+            if (CLEAR_DT) rec.setAttribute(DUPLICATE_TYPE_TAG, null);
```

Comment:
👍 I'll try to clean up wherever I see this if its something we want to move towards, all for it!